File: scripts/postgres_samples/tm_data.R
require(RPostgreSQL)

# database settings
default.db.name <- "transmart"
default.db.username <-"postgres"
default.db.password <-"postgres"
default.db.connection.host <-"localhost"
default.db.connection.port <- "5432"

# creating connection to postgreSQL database
get.connection <- function(db.connection.host=default.db.connection.host,
							db.connection.port=default.db.connection.port,
							db.username=default.db.username,
							db.password=default.db.password,
							db.name=default.db.name) {
	db.connection <- dbConnect(PostgreSQL(),
							host=db.connection.host,
							port=db.connection.port,
							user=db.username,
							password=db.password,
							dbname=db.name)
	# return connection object
	db.connection
}

# test function for getting patients' subject identifiers
get.patients.subjects <- function() {
	
	# getting database connection
	db.connection <- get.connection()
	
	# getting result set with studies
	rs <- dbSendQuery(db.connection, "SELECT DISTINCT patient_num, subject_id FROM tm_data.studies_patients")
	
	#fetching all data from result set as data.frame
	data <- fetch(rs,n=-1)
	
	# closing connection
	dbDisconnect(db.connection)
	
	# returning data
	data
}

# calling our function
get.patient.subjects()

File: scripts/postgres_samples/tm_data.m
% Enter your studyId
studyId = '<your studyId>';

% Database settings
% Enter correct database host, port
dbname = 'transmart'
username = 'postgres'
password = 'postgres'
host = '<host>'
port=5432

% SQL query to get data from database
sqlQuery = strcat('SELECT subject_id, data_value  FROM tm_data.studies_clinical_values WHERE study_id =''', studyId, ''' AND subject_id IS NOT NULL');

% Connecting to database
conn = database(dbname,username,password,...
				'Vendor','PostgreSQL',...
				'DriverType','thin',...
				'Server', host, 'PortNumber',port);
setdbprefs('DataReturnFormat','cellarray');

% Getting data from database (using cursor)
disp(sqlQuery);
curs = exec(conn, sqlQuery);
curs = fetch(curs);
sqlRes = cell2table(curs.Data);
close(curs);
close(conn);

% Converting SQL result to arrays
% Creating structures
sql_subjects = sqlRes{:,1}; % getting subject_ids

sql_data = sqlRes{:,2}; % getting values
sql_values = cellfun(@str2double, sql_data); % converting type of values to double